src/options.rs

```diff
@@ -409,15 +390,24 @@ impl Options {
             && atty::is(atty::Stream::Stdin);
 
         let mut builder = conn_params(&tmp.conn);
-        if let (Some(Command::Authenticate(auth)), None) = (&tmp.subcommand, &tmp.query) {
-            if builder.is_err() {
-                builder = Ok(Builder::new());
-                load_tls_options(&tmp.conn, builder.as_mut().unwrap())?;
+
+        // Special case for `edgedb instance link`
+        if let (Some(Command::Instance(i)), None) = (
+            &tmp.subcommand, &tmp.query
+        ) {
+            if let server::options::InstanceCommand::Link(
+                link
+            ) = &i.subcommand {
+                if builder.is_err() {
```

Comment:
This should have more specific error check. So `conn_params` function should emit a specific error whan `edgedb.toml` not found. You can look at `InstanceNotFound` error as an example of custom error and it's matching.

Reply:
I'm sorry but I'm not sure I get your idea here - in case `edgedb.toml` is not found, `edgedb instance link ...` should just ask about connection parameters, no?

Reply:
Sure but there are few ways how `conn_params` can fail (so `builder is Err(..)`). For example if you passed `edgedb --dsn bad-proto://something` user should see failure instead of running interactive mode.